cmake/Modules/FindTinyXML2.cmake

```diff
@@ -17,7 +17,9 @@ else()
   find_library(TINYXML2_LIBRARY tinyxml2)
 
   include(FindPackageHandleStandardArgs)
-  find_package_handle_standard_args(tinyxml2 DEFAULT_MSG TINYXML2_LIBRARY TINYXML2_INCLUDE_DIR)
+  find_package_handle_standard_args(tinyxml2
```

Comment:
We've been having offline discussions about whether or not this could be solved by changing `tinyxml2` to `TinyXML2`. This is what is done in tinyxml_vendor.

https://github.com/ros2/tinyxml_vendor/blob/2ea164ee7b868e01533bb09819e44ef0abbcb0e3/cmake/Modules/FindTinyXML.cmake#L71